packages/plugin-user/src/user-uuid-store.js

```diff
@@ -0,0 +1,102 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {patterns} from '@ciscospark/common';
+
+const usersByEmail = new WeakMap();
+const usersById = new WeakMap();
+
+/**
+ * @class
+ */
+export default class UserUUIDStore {
+  /**
+   * @constructs {UserUUIDStore}
+   */
+  constructor() {
+    usersByEmail.set(this, new Map());
+    usersById.set(this, new Map());
```

Comment:
Not exactly. It's a way to ensure private data for the class. the only way to access data in those maps is to call the corresponding methods on an instance of UserUUIDStore.

Basically, WeakMaps let us associate arbitrary data with an instance without modifying that instance (and therefor deoptimizing it).